### ISSUE TITLE
Dynamic DTO_FIELD within an oatpp::DTO

### ISSUE THREAD

**Issue by davidmainzer** [oatpp/oatpp#777]
Hi all,

is it possible to add DTO_FIELD dynamically during startup of the oatpp server? For example the number and type of the DTO_FIELD's are provided within a configuration. 

```
class SchemaData
   : public oatpp::DTO
{
   DTO_INIT( SchemaData, DTO );

   DTO_FIELD( String, type, "type" );                                                             
   DTO_FIELD( Object<SchemaDataProperties>, properties, "properties" );
};
```

So I don't know the exact number and type of the DTO_FIELD before we start the application thus i would like to add these fields dynamically. Unfortunately, these macros are not easy to replace with plain runtime extendable code. Exists a way to achieve a dynamic DTO creation process before we start the REST server? Even loading this information from an json, yaml or what ever would be a possible solution. Any hit is welcome.

Many thanks in advance.

**Comment by lganzzzo** [comment]
Hello @davidmainzer ,

The short answer, is yes - it's possible to create dynamic DTOs. 
Also, in deed, they are created dynamically in runtime during the startup.

Please give me some more info about the use-case, maybe there is a simpler solution.

The base class of all DTOs is [BaseObject](https://github.com/oatpp/oatpp/blob/master/src/oatpp/core/data/mapping/type/Object.hpp#L48)
You have to carefully create the correct object prototype and map all fields to real values.


Basically it's the way how [this experimental module](https://github.com/oatpp/oatpp-protobuf) was working.
It was dynamically creating DTO clones for protobuf objects

**Comment by davidmainzer** [comment]
Thanks for the fast answer :)

The endpoints i can already create dynamically but the structure of each endpoint is not known before. I will have a look inside the protobuf example but i saw it already.

My goal is to make the server response query api configurable during startup since we dont know before which fields of the endpoints we do support. Depending on the server configuration.
